cli/src/semgrep/core_runner.py

```diff
@@ -639,16 +641,10 @@ def plan_core_run(
         )
 
         lockfiles = target_manager.get_all_lockfiles()
-
-        rules = [
-            rule
-            for rule in rules
-            # filter out SCA rules with no relevant lockfiles
-            if not (isinstance(rule.product.value, out.SCA))
-            or any(lockfiles[ecosystem] for ecosystem in rule.ecosystems)
-        ]
```

Comment:
it looks like we're no longer filtering out not-applicable SCA rules before passing them to the core. What happens if the core gets a rule w/out a corresponding lock file?

Reply:
we now only filter in the parent caller now for consistency
 
see https://github.com/semgrep/semgrep/blob/f1719f3a137b169d24ab7b80be447e3aad762e98/cli/src/semgrep/scan_report.py#L318